src/lumigo_tracer/lambda_tracer/lambda_reporter.py

```diff
@@ -0,0 +1,293 @@
+import os
+import uuid
+import time
+import random
+import socket
+import datetime
+import http.client
+from pathlib import Path
+from typing import Optional, List, Dict, Any, Union
+from base64 import b64encode
+from functools import lru_cache
+
+from lumigo_tracer.lumigo_utils import (
+    InternalState,
+    get_logger,
+    Configuration,
+    EDGE_HOST,
+    aws_dump,
+    internal_analytics_message,
+    lumigo_safe_execute,
+    warn_client,
+    should_use_tracer_extension,
+    is_span_has_error,
+    get_region,
+)
+
+try:
+    import botocore
+    import boto3
+except Exception:
+    botocore = None
+    boto3 = None
+
+EDGE_PATH = "/api/spans"
+HTTPS_PREFIX = "https://"
+SECONDS_TO_TIMEOUT = 0.5
+EDGE_TIMEOUT = float(os.environ.get("LUMIGO_EDGE_TIMEOUT", SECONDS_TO_TIMEOUT))
+MAX_SIZE_FOR_REQUEST: int = int(os.environ.get("LUMIGO_MAX_SIZE_FOR_REQUEST", 1024 * 500))
+MAX_NUMBER_OF_SPANS: int = int(os.environ.get("LUMIGO_MAX_NUMBER_OF_SPANS", 2000))
+TOO_BIG_SPANS_THRESHOLD = 5
+NUMBER_OF_SPANS_IN_REPORT_OPTIMIZATION = 200
+COOLDOWN_AFTER_TIMEOUT_DURATION = datetime.timedelta(seconds=10)
+CHINA_REGION = "cn-northwest-1"
+LUMIGO_SPANS_DIR = "/tmp/lumigo-spans"
+
+edge_kinesis_boto_client = None
+edge_connection = None
+
+
+def establish_connection_global() -> None:
```

Comment:
This code was executed in global, and now is wrapped in a function (that is being called from `__init__`